### ISSUE TITLE
🔗 Menu widget slow loading (Duplicate of #21956, #24076)

### ISSUE THREAD

**Issue by gonzalo-terrahq** [elementor/elementor#25211]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

I'm using the new menu widget to create a menu menu with containers and custom content inside of it. The editor doesn't work fine and sometimes I can't save changes or try to use dynamic content from anchors and links.

I'm using some custom CSS and JS code, but I've tried deleting it and the issue remains. I've also tried deactivating all plugins, changing theme and deleting and reactivating Elementor + Elementor Pro. 

### Steps to reproduce

1. Create a header with new menu widget
2. Use dynamic content
3. Try to use internal links from dynamic content on anchors.

### Expected behavior

Expected behavior is a fast editor. Editor is going great on the rest of the pages and templates.

### Isolating the problem

- [X] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: Linux
	Software: nginx
	MySQL version: Percona Server (GPL), Release '27', Revision '2f8eeab2'$ v8.0.35-27
	PHP Version: 8.2.14
	PHP Memory Limit: 512M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 256M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.4.3
	Site URL: https://prosperopsdev.wpengine.com
	Home URL: https://prosperopsdev.wpengine.com
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 512M
	Max Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 3.0.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields PRO
		Version: 6.1.8
		Author: WP Engine

	Clickcease - Click Fraud Protection
		Version: 3.2.4
		Author: ClickCease

	Copy & Delete Posts
		Version: 1.4.6
		Author: Inisev

	Default featured image
		Version: 1.7.3
		Author: Jan Willem Oostendorp

	Elementor
		Version: 3.19.2
		Author: Elementor.com

	Elementor Pro
		Version: 3.19.2
		Author: Elementor.com

	Enable Media Replace
		Version: 4.1.5
		Author: ShortPixel

	Stop User Enumeration
		Version: 1.4.9
		Author: Fullworks

	Temporary Login Without Password
		Version: 1.8.3
		Author: StoreApps

	Ultimate Addons for Elementor
		Version: 1.36.29
		Author: Brainstorm Force

	WP Rocket
		Version: 3.15.9
		Author: WP Media

	Yoast SEO
		Version: 22.0
		Author: Team Yoast

	Yoast SEO Premium
		Version: 20.12
		Author: Team Yoast


== Must-Use Plugins ==
	Force Strong Passwords - WPE Edition
		Version: 1.8.0
		Author: Jason Cosper

	WP Engine Cache Plugin
		Version: 1.1.5
		Author: WP Engine

	WP Engine Seamless Login Plugin
		Version: 1.6.1
		Author: WP Engine

	WP Engine Security Auditor
		Version: 1.1.1
		Author: wpengine

	WP Engine System
		Version: 5.0.1
		Author: WP Engine


== Elements Usage ==
	
	container : 51
		accordion : 2
		button : 40
		container : 492
		counter : 3
		divider : 2
		heading : 291
		html : 17
		icon : 22
		icon-list : 13
		image : 28
		menu-anchor : 12
		nested-tabs : 6
		shortcode : 11
		text-editor : 104
	header : 5
		button : 4
		container : 75
		heading : 24
		image : 19
		mega-menu : 1
		text-editor : 15
		theme-site-logo : 1
	page : 20
		accordion : 2
		button : 47
		container : 388
		counter : 4
		divider : 7
		heading : 227
		html : 22
		icon : 27
		icon-list : 16
		image : 59
		menu-anchor : 11
		nested-tabs : 3
		shortcode : 5
		text-editor : 163
	popup : 8
		container : 1
		text-editor : 1
	wp-post : 0
		button : 6
		container : 60
		counter : 4
		heading : 28
		icon : 3
		image : 14
		lottie : 3
		media-carousel : 1
		text-editor : 23
		uael-faq : 1
	wp-page : 31
		accordion : 3
		button : 57
		container : 401
		counter : 4
		heading : 149
		html : 19
		icon : 20
		icon-list : 13
		image : 59
		menu-anchor : 12
		nested-tabs : 2
		shortcode : 2
		spacer : 1
		text-editor : 215
	loop-item : 8
		button : 1
		container : 2
		theme-post-featured-image : 1
		theme-post-title : 1


== Settings ==
	
	cpt_support: post, page, report, product
	disable_color_schemes: yes
	disable_typography_schemes: yes
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap


== Features ==
	Custom Fonts: 2
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Improved Asset Loading: Active
	Improved CSS Loading: Active by default
	Inline Font Icons: Active
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active by default
	Grid Container: Active
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Editor Top Bar: Inactive by default
	Optimized Gutenberg Loading: Active
	Build with AI: Active by default
	Landing Pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Active
	Optimize Image Loading: Active by default
	Global Style Guide: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Display Conditions: Inactive by default
	Form Submissions: Active by default
	Scroll Snap: Inactive
	Menu: Active
	Taxonomy Filter: Inactive by default


== Log ==
	
JS: showing 20 of 36JS: 2024-02-16 09:15:17 [error X 6][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view2269") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:15:17 [error X 6][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view2341") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view3628") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view3707") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view3766") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view3931") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view3990") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4081") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4147") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4213") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4279") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4345") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4411") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4528") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4654") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4720") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4846") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:43:27 [error X 3][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view4903") has already been destroyed and cannot be used. 
JS: 2024-02-16 09:48:47 [error X 1][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: "view10156") has already been destroyed and cannot be used. 
JS: 2024-02-16 10:21:08 [error X 6][../wp-content/plugins/elementor/assets/js/common-modules.min.js?ver=3.19.2:2:56156] Model.isValidChild() should be implemented, please provide 'isValidChild' functionality. 

PHP: showing 6 of 6PHP: 2024-02-16 09:20:58 [notice X 23][../wp-content/plugins/elementor/includes/base/controls-stack.php::228] hexdec(): Passing null to parameter #1 ($hex_string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ../wp-content/plugins/elementor/includes/base/controls-stack.php(228): class type hexdec()
#2: ../wp-content/plugins/elementor/includes/base/controls-stack.php(242): Elementor\Controls_Stack -> get_id_int()
#3: ../wp-content/plugins/elementor-pro/modules/mega-menu/widgets/mega-menu.php(1979): Elementor\Controls_Stack -> get_widget_number()
#4: ../wp-content/plugins/elementor-pro/modules/mega-menu/widgets/mega-menu.php(2027): ElementorPro\Modules\MegaMenu\Widgets\Mega_Menu -> widget_number()
',
)]
PHP: 2024-02-16 09:20:58 [notice X 23][../wp-content/plugins/elementor/includes/base/controls-stack.php::228] hexdec(): Passing null to parameter #1 ($hex_string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ../wp-content/plugins/elementor/includes/base/controls-stack.php(228): class type hexdec()
#2: ../wp-content/plugins/elementor/includes/base/controls-stack.php(242): Elementor\Controls_Stack -> get_id_int()
#3: ../wp-content/plugins/elementor-pro/modules/mega-menu/widgets/mega-menu.php(1979): Elementor\Controls_Stack -> get_widget_number()
#4: ../wp-content/plugins/elementor-pro/modules/mega-menu/widgets/mega-menu.php(2028): ElementorPro\Modules\MegaMenu\Widgets\Mega_Menu -> widget_number()
',
)]
PHP: 2024-02-16 15:28:25 [warning X 48][../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key "condition_type" [array (
  'trace' => '
#0: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: ../wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2024-02-20 11:11:35 [warning X 16][../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key "condition_type" [array (
  'trace' => '
#0: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: ../wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2024-02-20 11:11:56 [notice X 2][../wp-content/plugins/elementor/includes/base/controls-stack.php::228] hexdec(): Passing null to parameter #1 ($hex_string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ../wp-content/plugins/elementor/includes/base/controls-stack.php(228): class type hexdec()
#2: ../wp-content/plugins/elementor/includes/base/controls-stack.php(242): Elementor\Controls_Stack -> get_id_int()
#3: ../wp-content/plugins/elementor-pro/modules/mega-menu/widgets/mega-menu.php(1979): Elementor\Controls_Stack -> get_widget_number()
#4: ../wp-content/plugins/elementor-pro/modules/mega-menu/widgets/mega-menu.php(2027): ElementorPro\Modules\MegaMenu\Widgets\Mega_Menu -> widget_number()
',
)]
PHP: 2024-02-20 11:11:56 [notice X 2][../wp-content/plugins/elementor/includes/base/controls-stack.php::228] hexdec(): Passing null to parameter #1 ($hex_string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ../wp-content/plugins/elementor/includes/base/controls-stack.php(228): class type hexdec()
#2: ../wp-content/plugins/elementor/includes/base/controls-stack.php(242): Elementor\Controls_Stack -> get_id_int()
#3: ../wp-content/plugins/elementor-pro/modules/mega-menu/widgets/mega-menu.php(1979): Elementor\Controls_Stack -> get_widget_number()
#4: ../wp-content/plugins/elementor-pro/modules/mega-menu/widgets/mega-menu.php(2028): ElementorPro\Modules\MegaMenu\Widgets\Mega_Menu -> widget_number()
',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatibility not specified
	Ultimate Addons for Elementor: Compatibility not specified

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by nicholaszein** [comment]
Hello @gonzalo-terrahq!

Thank you for your submission! 🙏 

### 🔗 🐞 This issue is a Duplicate of #21956, and #24076

Which are on the same subject. 

For this reason, I'm closing this thread and 🔃 **Merging** it with the original, so we can have all the discussions in one location, and deal with it more efficiently.

Feel free to continue the conversation there. 😉 

### 🔍 Please, in the future, remember to check if there isn't any thread similar to yours already opened. 

Kind regards

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved with the release of Elementor v3.23.0! 🥳 

The time to interaction should improve by 60%-98%, and performance overall should be better. For a full breakdown please read [this article](https://developers.elementor.com/elementor-3-23-developers-update/#:~:text=Performant%20Nested%20Elements%20In%20the%20Editor).

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
